pkg/handler/repositories.go

```diff
@@ -9,14 +9,24 @@ import (
 	"github.com/content-services/content-sources-backend/pkg/dao"
 	"github.com/content-services/content-sources-backend/pkg/db"
 	"github.com/content-services/content-sources-backend/pkg/models"
+	"github.com/davecgh/go-spew/spew"
 	"github.com/labstack/echo/v4"
 	"github.com/redhatinsights/platform-go-middlewares/identity"
 )
 
-func RegisterRepositoryRoutes(engine *echo.Group) {
+type RepositoryHandler struct {
+	RepositoryDao dao.RepositoryDao
+}
+
+func RegisterRepositoryRoutes(engine *echo.Group, rDao *dao.RepositoryDao) RepositoryHandler {
+	rh := RepositoryHandler{RepositoryDao: *rDao}
 	engine.GET("/repositories/", listRepositories)
+	engine.GET("/repositories/:uuid", rh.fetch)
+	engine.PUT("/repositories/:uuid", rh.fullUpdate)
+	engine.PATCH("/repositories/:uuid", rh.partialUpdate)
 	engine.DELETE("/repositories/:uuid", deleteRepository)
-	engine.POST("/repositories/", createRepository)
+	engine.POST("/repositories/", rh.createRepository)
+	return rh
```

Comment:
you're right, I think its not explicitly needed to be passed in or returned for the current tests.  I think it made sense for me to pass in the dao from a central place rather than have each handler instantiate the dao itself.  

If we move to these style tests:  https://github.com/content-services/content-sources-backend/blob/main/pkg/handler/api_test.go#L15-L36  it probably would need to be passed in at least, but not returned.   I'll remove the return but would probably prefer to keep it being passed in unless you object

Reply:
We can leave it as being passed in. I would also just change the name here from dao to rDao, so it doesn't conflict with the import: https://github.com/content-services/content-sources-backend/blob/5390ddc35a58951c711ca0bb346ea29cb9bf35c1/pkg/handler/api.go#L51-L52

Reply:
I imagine in the future, we could have a 'master dao' object, that you could pass around too instead of a dao object for each type, but another thought for another day